### PR TITLE
users: patron in multiple organisations

### LINKED ISSUE #1460: A user has an account in multiple organisations.
**How it works**

The organisations shared the RERO ID of the users. Only one organisation can create a user record linked to a specific RERO ID.

**Improvement suggestion**
Each organisation should be able to create a user resource linked to a RERO ID.

3 impacts:
* In the patron editor, it is possible to attach the patron to a RERO ID, only if it has no patron attached from this organisation, and even if it already has patrons attached from other organisations.
* In the patron account of the public view, the transactions are displayed according to the organisation
   * either they are filtered on each organisation view, and mixed on the global view
   * **[preferred by PO]** or they are mixed in all views (with ev. select to filter by org., can be done later)
   * or there is a select to choose the organisation for which the transactions must be displayed
* In the global public view, the request action might be adapted for patrons in multiple organisations.

### LINKED ISSUE #1490: No feedback is given to the user if the email is not confirmed
**Describe the bug**

On the login page if the email is not confirmed no warning or error message is given to the user.

**To Reproduce**

1. Create a fresh RERO ID account
2. Logout
3. Try to login without having confirmed the email address

**Expected behavior**

An error message should be displayed to inform the user that he has to confirm his email address.

Version `v.014.1`

### LINKED ISSUE #1634: The affiliation library disappears in some cases in the patron editor
**Describe the bug**

The affiliation library disappears in some cases in the patron editor. When it happens, it is then impossible to save the record. This use case might apply to all editors.

**To Reproduce**

1. Open the editor to create a new patron
2. Add an affiliation library without actually selecting a library
3. Change the default "patron" role to "librarian"
4. Come back to the "patron" role
5. See the field "Affiliation libraries": the sub-fields and the option to add them disappeared

**Expected behavior**

It is possible to add this field.

**Context**

* server: [ils.test.rero.ch][1]
* version: `v0.15.0`

[1]: https://ils.test.rero.ch

**Screenshots**

image

### LINKED ISSUE #1804: Keep history settings enabled but not taken into account
**Describe the bug**

As a user, even if I change the 'keep history' settings, my loan history is not saved.
In my personal data tab, it still displays 'The loan history is not saved'.
It is the same as a librarian. I can modify the setting for a user but it is not taken into account.
Note: The 'keep history' settings is displayed as enabled in the edition form of the public and the professional interface.

**To Reproduce**

1. Log is as a patron.
2. Go to 'My account'
3. Modify the personal data.

Or modify the history settings as a librarian and make some loan transactions.

**Expected behavior**

The user should see a loan history tab.

**Context**

* server: [ils.test.rero.ch][1] or [ilsdev.test.rero.ch][2]
* version: `v01.1.0` or the commit hash (see frontpage).
* browser: [if relevant, e.g. chrome, safari, version 78].

[1]: https://ils.test.rero.ch
[2]: https://ilsdev.test.rero.ch

**Screenshots**

Personal data tab:
image

Settings enabled:
image

### LINKED ISSUE #1837: public view: Menu "My Account" doesn't appear for user with role librarian
**Describe the bug**

When a user with profile librarian and patron is logged, he can't access its patron account in the public view.
The menu "My Account" is not displayed

The problem is limited to user with role patron AND librarian

**To Reproduce**

1. Go to ils.test.rero.ch
2. login with reroilstest+jean@gmail.com
3. In the public view, click on the green button
4. See you can't not find "My Account"

**Expected behavior**
 
A user with role patron AND librarian must be able to access its patron account in the public view.

* server: ils.test.rero.ch 
* version: `v1.1.0`

### LINKED ISSUE #1851: Requests: Waiting queue position not displayed anymore (public interface)
**Describe the bug**

In the public interface, a user cannot see anymore his position in the waiting queue.

**To Reproduce**

1. Log in as a librarian.
2. Add several requests to an item.
3. Log in as a user

**Expected behavior**

As a user, I should be able to see in my account 'request' tab, my posititon in the queue if there are several requests on the item.

**Context**

* server: [ils.test.rero.ch][1] or [ilsdev.test.rero.ch][2]?
* version: `v0.1.0.0` or the commit hash (see frontpage).
* browser: [if relevant, e.g. chrome, safari, version 78].

[1]: https://ils.test.rero.ch
[2]: https://ilsdev.test.rero.ch

**Screenshots**

image

### PR DESCRIPTION
**Pull request by jma**
* Replaces the string `current_patron` by `current_librarian` in the
  entire project.
* Adds a new `current_patrons` containing the list of the logged user
  patron accounts.
* Removes useless jinja detailed view configuration and templates.
* Fixes link redirection for the ill request.
* Cleans the patron request Web API.
* Fixes tests and permissions.
* Replaces `current_organisation.pid` by `current_librarian.organsiation_pid.
* Fixes tests and premissions.
* Gets the patron request rank given a patron instead of a barcode.
* Fixes translated string on circulation.
* Hides the profile menu given the right roles.
* Adds organisations information in the User API.
* Closes #1837.
* Closes #1460.
* Closes #1851.
* Closes #1804.
* Closes #1634.
* Closes #1490.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>
Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Why are you opening this PR?

- https://tree.taiga.io/project/rero21-reroils/us/2034?milestone=286652

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* https://github.com/rero/rero-ils-ui/pull/567

## For Reviewers
This huge PR contains 4 commits, you can use the followings links to make you review on each commit to make your job easier:

- https://github.com/rero/rero-ils/pull/1797/commits/94db328e99a35702f0e9d6099f43015a66457dba, is just a find and replace: `current_patron` -> `current_librarian`
- https://github.com/rero/rero-ils/pull/1797/commits/d5e1024f040cc4eed55fd5f3f1929da63823ada6, adds a new concept `current_patrons` which is the list of current patron account (one max per organisation) related to the current logged user.
- https://github.com/rero/rero-ils/pull/1797/commits/86de28071fe7a8ec3fa2df2bfc9e9df9bf41d42b, this is basically a refactoring commit by removing a useless `current_organisation` proxy.
- https://github.com/rero/rero-ils/pull/1797/commits/1af64a01fde0f863fd7e498795e6a3b7cb7e56e3, small fix for the UI interface.

## How to test?

- Simonetta is a patron in the two organisation 1 and 2
- Go to the admin view and try to create one of each type of resource (access rights)
- Check that the search list of each resource is filtered correctly.
- Try to attach a patron of an other organisation to your organisation and try to do a request, validate a request make checkin/checkout for this new patron. Check the patron profile of the newly created patron.
- Try to create a new librarian in your organisation which is already a librarian in an other org: you should not be able to do it.
- Try to generate a notification for a user with multiple patrons.
- Register a new user from the user interface and create a patron account.
- Change the patron information in the org1 et see if it is reported to the org2.
- Try to change the password for a user with two organisations.
- Check that the reset password instruction does not contains the library link.
- Check the patron request with items of two different organisations.
- Try to change the role of a patron with two organisations, the detailed view and the editor should contains the right roles.
- Try to change the "keep history" button in the patron profile and see that it is correction.
- Change the patron personal information in the patron profile and check that all the patrons informations are up to date in the professional views.
- Check the profile entry menu and ill request menu is correct for a librarian(hidden), patron and a user with a multiple patron accounts.
- etc.

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: rero_ils/modules/patrons/api.py [hunk 1]
```diff
@@ -322,10 +339,16 @@ def get_patron_by_user(cls, user):
...
         """Get all patrons by user."""
         if hasattr(user, 'id'):
-            result = PatronsSearch()\
-                .filter('term', user_id=user.id)\
-                .source(includes='pid').scan()
-            return [cls.get_record_by_pid(record.pid) for record in result]
+            result = PatronsSearch().filter(
+                'term',
+                user_id=user.id
+            ).source(includes='pid').scan()
```
Comment (zannkukai) [inline rero_ils/modules/patrons/api.py:345]: Old presentation seems more readable for me. Design choice

**Review by lauren-d — APPROVED** [review]

**Comment by ManaDeweerdt** [comment]
[ ] A request spins forever. Ex: try to request as Simonetta https://ilsdev.test.rero.ch/global/documents/2000409

**Review by pronguen — COMMENTED** [review]
- [ ] In the document editor, impossible to look for a person (field contribution)
- [x] **(!)** This [person](https://ilsdev.test.rero.ch/professional/circulation/patron/nicolas/loan) is librarian in org1, patron in org1 and patron in org2. He has a loan in org2 that is visible from the professional interface, but not shown in the public account.
- [x] **(!)** I edited a user (street number) from org1, it works. But in org2, the old street number is still shown in the pro [patron profile](https://ilsdev.test.rero.ch/professional/circulation/patron/giulia/profile), see screenshot. I refreshed the page without cache but it did not help. When I edit the user from org2, in the user editor, the right street number is displayed. The same happens with "Keep history".
image
- [ ] The reset password instruction does not contains the library link (ok!) but it ends weirdly without signature. We could add "Votre bibliothèque / Your library / Ihre Bibliothek / La sua biblioteca" as a signature?
image
- [x] After editing my personal data as Simonetta (from public view), I land on an internal error server. But the changes were saved.
    - [ ] As mentioned previously, this changed is well displayed the pro interface (https://ilsdev.test.rero.ch/professional/circulation/patron/2050124311/profile) of one org but not of the other org.

**Review by iGormilhit — APPROVED** [review]
Commit message approved.

**Pull request merged**